modules/spring-config/src/main/resources/META-INF/mule.xsd

```diff
@@ -8109,6 +8121,115 @@
         </xsd:union>
     </xsd:simpleType>
 
+
+    <xsd:simpleType name="expression">
+        <xsd:annotation>
+            <xsd:documentation>
+                A custom xsd:string-like type that represents a MEL expression.
+                Attributes should use this type to declare that they support expressions.
+                It allows property placeholders and requires the value to not be blank.
+            </xsd:documentation>
+        </xsd:annotation>
+        <xsd:union>
+            <xsd:simpleType>
+                <xsd:restriction base="propertyPlaceholderType"/>
+            </xsd:simpleType>
+            <xsd:simpleType>
+                <xsd:restriction base="nonBlankString"/>
+            </xsd:simpleType>
+        </xsd:union>
+    </xsd:simpleType>
+
+    <xsd:simpleType name="expressionString">
+        <xsd:annotation>
+            <xsd:documentation>
+                An embeddable expression that outputs a String
+            </xsd:documentation>
+        </xsd:annotation>
+        <xsd:restriction base="expression" />
+    </xsd:simpleType>
+
+    <xsd:simpleType name="expressionBoolean">
+        <xsd:annotation>
+            <xsd:documentation>
+                A non-embeddable expression that outputs a Boolean
+            </xsd:documentation>
+        </xsd:annotation>
+        <xsd:restriction base="expression" />
+    </xsd:simpleType>
+
+    <xsd:simpleType name="expressionInt">
```

Comment:
yes, the idea is to start using it every chance we get in order to hint studio about things (Studio can't take advantage of that right now, but we talked to them about doing it in the future)